metaflow/plugins/kfp/kfp_pod_decorator.py

```diff
@@ -0,0 +1,27 @@
+from metaflow.decorators import StepDecorator
+
+
+class PodAnnotationDecorator(StepDecorator):
+    """
+    Add a single pod annotation. This decorator can be used multiple time per step.
+    Repeatedly assigning value under same key will overwrite previous value.
+    """
+
+    name = "pod_annotation"
+    defaults = {
+        "key": None,
+        "value": None,
+    }
+
+
+class PodLabelDecorator(StepDecorator):
+    """
+    Add a single pod label. This decorator can be used multiple time per step.
+    Repeatedly assigning value under same key will overwrite previous value.
+    """
+
+    name = "pod_label"
+    defaults = {
+        "key": None,
+        "value": None,
+    }
```

Comment:
Because these are KFP/k8s specific I suggest to have pod annotation and labels as part of a `@kfp` decorator as:
```python
@kfp(labels=[], annotations=[])
```

Reply:
Can do, and it probably will be less complex code.